### PR TITLE
Disable projections when ARRAY JOIN is used.

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Disable projections when `ARRAY JOIN` is used. Currently projection analysis will break array join aliases.

Detailed description / Documentation draft:
.

### CODE REVIEW

File: src/Storages/ProjectionsDescription.cpp [hunk 1]
```diff
@@ -217,23 +211,12 @@ ProjectionDescription::getMinMaxCountProjection(const ColumnsDescription & colum
...
-    }
     result.type = ProjectionDescription::Type::Aggregate;
     StorageInMemoryMetadata metadata;
     metadata.setColumns(ColumnsDescription(result.sample_block.getNamesAndTypesList()));
-    metadata.partition_key = KeyDescription::getSortingKeyFromAST({}, metadata.columns, query_context, {});
-    metadata.sorting_key = KeyDescription::getSortingKeyFromAST({}, metadata.columns, query_context, {});
-    metadata.primary_key = KeyDescription::getKeyFromAST({}, metadata.columns, query_context);
+    metadata.partition_key = KeyDescription::getSortingKeyFromAST({}, {}, query_context, {});
+    metadata.sorting_key = KeyDescription::getSortingKeyFromAST({}, {}, query_context, {});
+    metadata.primary_key = KeyDescription::getKeyFromAST({}, {}, query_context);
```
Comment (KochetovNicolai) [inline src/Storages/ProjectionsDescription.cpp:219]: It is confusing that those functions are called, well, on nothing.
Maybe default constructor for KeyDescription will do? (or maybe add one from context)

File: src/Storages/ProjectionsDescription.cpp [hunk 2]
```diff
@@ -125,31 +123,16 @@ ProjectionDescription::getProjectionFromAST(const ASTPtr & definition_ast, const
...
-    {
-        for (const auto & key : select.getQueryAnalyzer()->aggregationKeys())
-            result.sample_block_for_keys.insert({nullptr, key.type, key.name});
-    }
-
-    for (size_t i = 0; i < result.sample_block.columns(); ++i)
-    {
-        const auto & column_with_type_name = result.sample_block.getByPosition(i);
-
-        if (column_with_type_name.column && isColumnConst(*column_with_type_name.column))
```
Comment (KochetovNicolai) [inline src/Storages/ProjectionsDescription.cpp:139]: Is this check not needed anymore?

Comment (amosbird) [reply src/Storages/ProjectionsDescription.cpp:139]: It's been moved down below.

File: src/Storages/ProjectionsDescription.cpp (hunk 1 above)
Comment (amosbird) [reply src/Storages/ProjectionsDescription.cpp:219]: Indeed. I'll make a proper constructor for this.

File: src/Storages/ProjectionsDescription.cpp (hunk 2 above)
Comment (KochetovNicolai) [reply src/Storages/ProjectionsDescription.cpp:139]: 've seen it.

**Review by KochetovNicolai — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@KochetovNicolai PR should not be merged with failed tests.
If the failures are unrelated, first classificate and fix these failures.

**Pull request merged**